### PR TITLE
add getproperty and propertynames for data frame iterators

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/1895.

CC @oschulz

### CODE REVIEW

File: src/abstractdataframe/iteration.jl [hunk 1]
```diff
@@ -75,12 +75,16 @@ x  3
 eachrow(df::AbstractDataFrame) = DataFrameRows(df, index(df))
 
 Base.IndexStyle(::Type{<:DataFrameRows}) = Base.IndexLinear()
-Base.size(itr::DataFrameRows) = (size(itr.df, 1), )
+Base.size(itr::DataFrameRows) = (size(parent(itr), 1), )
 
 Base.@propagate_inbounds Base.getindex(itr::DataFrameRows, i::Int) =
-    DataFrameRow(itr.df, itr.index, i)
+    DataFrameRow(parent(itr), getfield(itr, :index), i)
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]: Define an `index` method for `DataFrameRows`?

File: test/iteration.jl [hunk 2]
```diff
@@ -115,4 +115,17 @@ end
     @test copy(erd[1]) == (y1 = 51, y2 = 21, y3 = 31) # the removed columns are reflected
 end
 
+@testset "getproperty and propertynames" begin
+    df_base = DataFrame([11:16 21:26 31:36 41:46])
+    for df in [df_base, view(df_base, 1:3, 1:3)]
```
Comment (nalimilan) [inline test/iteration.jl]:
```suggestion
    for df in (df_base, view(df_base, 1:3, 1:3))
```

File: test/iteration.jl [hunk 3]
```diff
@@ -115,4 +115,17 @@ end
...
+        for x in [eachcol(df), eachcol(df, true), eachrow(df)]
```
Comment (nalimilan) [inline test/iteration.jl]:
```suggestion
        for x in (eachcol(df), eachcol(df, true), eachrow(df))
```

File: src/abstractdataframe/iteration.jl (hunk 1 above)
Comment (bkamins) [reply src/abstractdataframe/iteration.jl]: Actually maybe we should remove the `:index` field. We currently do not use it. It is always set to `index(df)` and we do not export `index` anyway.

Comment (nalimilan) [reply src/abstractdataframe/iteration.jl]: OK, simplification is always nice.

Comment (bkamins) [reply src/abstractdataframe/iteration.jl]: I have changed this - let us see how it goes.

**Comment by bkamins** [comment]
This should be good to have a look at.

**Pull request merged**